pkg/flowaggregator/flowaggregator.go

```diff
@@ -724,3 +818,130 @@ func (fa *flowAggregator) createInfoElementForTemplateSet(ieName string, enterpr
 	}
 	return ie, nil
 }
+
+// updateIfConfigChanges has a watcher to detect if modification happens. Then check whether we should update clickhouse or flow-collector at runtime
+func (fa *flowAggregator) updateIfConfigChanges(stopCh <-chan struct{}) {
+	if fa.configData == nil {
+		var err error
+		fa.configData, err = ioutil.ReadFile(fa.configFile)
+		if err != nil {
+			klog.ErrorS(err, "Get flow aggregator configData failed:")
+		}
+	}
+	go wait.NonSlidingUntil(fa.watchConfigmap, 5*time.Second, stopCh)
+}
+
+func (fa *flowAggregator) watchConfigmap() {
+	klog.Info("Watching for FA configuration file")
+	defer klog.Info("Stopping watch on FA configuration file")
+	watcher, err := fsnotify.NewWatcher()
+	if err != nil {
+		klog.ErrorS(err, "Error when creating watcher")
+	}
```

Comment:
IMO, this is a fatal error. We should return the error, and exit the FA process on start up if we fail to create the initial watch.

Reply:
Make sense, I've changed it to klog.Fatal